sklearn/neighbors/_base.py

```diff
@@ -770,12 +775,18 @@ class from an array representing our data set and ask who's
                 parallel_kwargs = {"backend": "threading"}
             else:
                 parallel_kwargs = {"prefer": "threads"}
-            chunked_results = Parallel(n_jobs, **parallel_kwargs)(
-                delayed(_tree_query_parallel_helper)(
-                    self._tree, X[s], n_neighbors, return_distance
+
+            with config_context(assume_finite=True):
+                # We remove the validation of the query points
+                # (in *parallel_kwargs) done at the beginning of
+                # BinaryTree.query as those points already got
+                # validated in the caller.
```

Comment:
Checks running time (represented by the small spike at the beginning) here is negligible.
![Screenshot 2021-08-06 at 18-05-07 Screenshot](https://user-images.githubusercontent.com/13029839/128539511-b8ce77d5-6252-4d8b-bbb5-c99396d8c5c7.png)

<details>

<summary> Script </summary>

```python
# profile.py
from sklearn.datasets import make_regression
from sklearn.neighbors import NearestNeighbors


def main(args=None):
    X_train, _ = make_regression(n_samples=100_000, n_features=10)
    X_test, _ = make_regression(n_samples=100_000, n_features=10)

    nn = NearestNeighbors(algorithm='kd_tree').fit(X_train)
    nn.kneighbors(X_test, n_neighbors=2)


if __name__ == "__main__":
    main()

```

```sh
giltracer --state-detect profile.py
```

</details>